pydantic_core/core_schema.py

```diff
@@ -1447,16 +1492,77 @@ def __call__(self, __input_value: Any, __info: ValidationInfo) -> Any:  # pragma
         ...
 
 
+class ModelFieldValidatorFunction(Protocol):
+    def __call__(self, __input_value: Any, __info: ModelFieldValidationInfo) -> Any:  # pragma: no cover
+        ...
+
+
+class ModelFieldValidatorFunctionSchema(TypedDict):
+    call: Required[ModelFieldValidatorFunction]
+    type: Literal['method']
+
+
+class ValidatorFunctionSchema(TypedDict):
+    call: Required[ValidatorFunction]
+    type: Literal['function']
+
+
 class FunctionSchema(TypedDict, total=False):
+    function: Required[Union[ModelFieldValidatorFunctionSchema, ValidatorFunctionSchema]]
     type: Required[Literal['function']]
     mode: Required[Literal['before', 'after']]
-    function: Required[ValidatorFunction]
     schema: Required[CoreSchema]
     ref: str
     metadata: Any
     serialization: SerSchema
 
 
+def model_field_function_before_schema(
```

Comment:
```suggestion
def method_before_schema(
```
1. fewer words
2. should work for dataclasses too

Same below.

Reply:
Is `method`/`function` really the right distinction (I realize I was the one that introduced it). Consider:

```python
class Model:
  @validator
  def val_x(cls, val, info):  val
```

This is technically a _bound class method_ or a _method bound to the class_. But we'd be putting it down the `function` code path. Which is a bit confusing. In general I think the nomenclature of `function` vs. `method` in Python is confusing. So I think perhaps it's best to differentiate between a validator attached to a model field vs. not. And by `model` here I mean `model` in the broadest sense to include dataclasses, TypedDict, etc. So maybe something like `field_before_validation_callback` and `before_validation_callback` would be good names?

Reply:
That is fine by me, let's confirm on call before I rename for the 5th time 🤣

Reply:
Big rename in 3b91441